internal/sbi/processor/pdu_session.go

```diff
@@ -673,6 +654,166 @@ func (p *Processor) HandlePDUSessionSMContextUpdate(
+					DLPDR.FAR.State = smf_context.RULE_REMOVE
+
+					pdrList = append(pdrList, ULPDR)
+					farList = append(farList, ULPDR.FAR)
+
+					pdrList = append(pdrList, DLPDR)
+					farList = append(farList, DLPDR.FAR)
+				}
+			}
+
+			// re-initialize DCTunnel
+			smf_context.ReleaseTEID(dcTunnel.ANInformation.TEID)
+			smContext.DCTunnel = smf_context.NewUPTunnel()
+			smContext.NrdcIndicator, sendPFCPModification = false, true
+
+			smContext.Log.Infoln("Release DCTunnel")
+		} else {
+			smContext.Log.Infof("Only master tunnel is active - Tunnel TEID: %d", smContext.Tunnel.ANInformation.TEID)
+
+			// handle the PDU session resource modify indication transfer message
+			if err = smf_context.
+				HandlePDUSessionResourceModifyIndicationTransfer(body.BinaryDataN2SmInformation, smContext); err != nil {
+				smContext.Log.Errorf("Handle PDUSessionResourceModifyIndicationTransfer failed: %+v", err)
+				sendPFCPModification = false
+			} else {
+				// apply the rules on DCTunnel
+				if err = smContext.ApplyDcPccRulesOnDcTunnel(); err != nil {
+					smContext.Log.Errorf("ApplyDcPccRulesOnDcTunnel failed: %+v", err)
+					sendPFCPModification = false
+				} else {
+					// append the rules on DCTunnel
+					for _, dataPath := range dcTunnel.DataPathPool {
+						if dataPath.Activated {
+							ANUPF := dataPath.FirstDPNode
+							ULPDR := ANUPF.UpLinkTunnel.PDR
+							DLPDR := ANUPF.DownLinkTunnel.PDR
+							if DLPDR.Precedence == 255 {
+								continue
+							}
+
+							ULPDR.FAR.ApplyAction = pfcpType.ApplyAction{
+								Buff: false,
+								Drop: false,
+								Dupl: false,
+								Forw: true,
+								Nocp: false,
+							}
+							DLPDR.FAR.ApplyAction = pfcpType.ApplyAction{
+								Buff: false,
+								Drop: false,
+								Dupl: false,
+								Forw: true,
+								Nocp: false,
+							}
+
+							DLPDR.State = smf_context.RULE_INITIAL
+							DLPDR.FAR.State = smf_context.RULE_INITIAL
+							ULPDR.State = smf_context.RULE_INITIAL
+							ULPDR.FAR.State = smf_context.RULE_INITIAL
+
+							pdrList = append(pdrList, DLPDR)
+							farList = append(farList, DLPDR.FAR)
+
+							pdrList = append(pdrList, ULPDR)
+							farList = append(farList, ULPDR.FAR)
+						}
+					}
+
+					// remove the rules for specified QoS traffic on Tunnel
+					for _, dataPath := range tunnel.DataPathPool {
+						if dataPath.Activated {
+							ANUPF := dataPath.FirstDPNode
+							DLPDR := ANUPF.DownLinkTunnel.PDR
+							if DLPDR.Precedence == 255 {
+								continue
+							}
+
+							DLPDR.State = smf_context.RULE_REMOVE
+							DLPDR.FAR.State = smf_context.RULE_REMOVE
+
+							pdrList = append(pdrList, DLPDR)
+							farList = append(farList, DLPDR.FAR)
+						}
+					}
+					smContext.NrdcIndicator, sendPFCPModification = true, true
+
+					smContext.Log.Infoln("Activate DCTunnel")
+				}
+			}
+		}
+
+		if sendPFCPModification {
+			smContext.SetState(smf_context.PFCPModification)
+			switch smContext.NrdcIndicator {
```

Comment:
smContext.NrdcIndicator's type is boolean.
Please use `if-else` statement instead of a `switch case` statement.